web_widget_digitized_signature/static/src/js/digital_sign.js

```diff
@@ -0,0 +1,141 @@
+            var images={};
+            var url;
+            if (this.get('value') && !utils.is_bin_size(this.get('value'))) {
+                url = 'data:image/png;base64,' + this.get('value');
+            }else if (this.get('value')) {
+                var id = JSON.stringify(this.view.datarecord.id || null);
+                self.digita_dataset = new data.DataSetSearch(self, self.view.model, {}, []);
+                self.digita_dataset.read_slice(['id', self.name], {'domain': [['id', '=', id]]}).then(function(records){
+                    _.each(records,function(record){
+                        if(record[self.name]){
+                            images[self.name] = record[self.name]
+                        }else{
+                            images[self.name] = ""
+                        }
+                    })
+                })
+                var field = this.name;
+                if (this.options.preview_image)
+                    field = this.options.preview_image;
+                url = this.session.url('/web/binary/image', {
+                          model: this.view.dataset.model,
+                          id: id,
+                          field: field,
+                          t: (new Date().getTime()),
+                });
+            }else {
+                images[self.name] = ""
+                url = this.placeholder;
+                self.set('value',images[self.name])
+            }
+            var $img = $(QWeb.render("FieldBinaryImage-img", { widget: this, url: url }));
+            this.$el.find('img').remove();
+            var sign_options = {};
+            if ('width' in self.node.attrs){
+                sign_options.width = self.node.attrs.width;
+            }
+            if ('height' in self.node.attrs){
+                sign_options.height = self.node.attrs.height;
+            }
+            var actual_mode = this.view.get("actual_mode");
+            if(actual_mode !== 'edit' && actual_mode !== 'create'){
+                this.$el.prepend($img);
+            }else if(actual_mode == 'edit' || actual_mode == 'create'){
+                if( actual_mode == 'create'){
+                    images = {};
+                }
+                this.$el.find('> img').remove();
+                this.$el.find('> canvas').remove();
+                if(! this.get('value')){
+                    this.$el.find('> img').remove();
+                    this.$el.find(".signature").signature(sign_options);
+                }else if(this.get('value')){
+                    this.$el.prepend($img);
+                }
+            }
+            this.$el.find('.clear_sign').click(function(){
+                if(self.$el.find('> img').length == 0){
+                    images[self.name] = ""
+                    self.$el.find(".signature").show();
+                    self.$el.find(".signature").signature('clear');
+                }
+            });
+
+            $('.save_sign').click(function(){
```

Comment:
You mostly want to use `this.$` or `this.$el.find` instead of plain `$` to avoid these things affecting elements outside its element. Same happens in other places.

Also big methods bound to these elements would be much better if put as methods for this class, instead of scoped lambdas, to allow proper inheritance.